how-to/hints-and-tips/docs/appassets-and-launch-external-process.md

```diff
@@ -0,0 +1,279 @@
+
+## A deeper dive into App Assets and Launch External Process
+
+### App Assets
+
+App Assets is a way of defining something you wish to package and make available through your OpenFin application. Generally these are applications of one kind or another. App Assets can be defined in two ways.
+
+#### App Assets via Manifest
+
+In your manifest file (this is a json file that describes your OpenFin application and is hosted on your server) you can add an appAssets setting which defines one or more App Assets that should be fetched as part of your application.
+
+This snippet is taken from our website: [Launch Assets By Using A Secured API](https://developers.openfin.co/of-docs/docs/launch-assets-by-using-a-secured-api#launch-application-assets)
+
+```json
+"appAssets": [
+    {
+        "src": "http://filesamples.com/exe.zip",
+        "alias": "myApp",
+        "version": "4.12.8",
+        "target": "myApp.exe",
+        "args": "a b c d"
+    },
+    {
+        "src": "http://examples.com/exe.zip",
+        "alias": "myApp2",
+        "version": "5.12.8",
+        "target": "myApp2.exe",
+        "args": "a b c",
+        "mandatory": false
+    }
+]
+```
+
+The definition of an app asset and what each property means can be found here: [OpenFin.AppAssetInfo](https://developer.openfin.co/docs/javascript/stable/interfaces/OpenFin.AppAssetInfo.html).
+
+Both apps provide a source url, a version number for the zipped file, an alias which can be used for launching it, a target to indicate what file should be launched within the zip file once extracted and what command line args should be passed when calling it.
+
+The main difference between the two is that the first entry will cause the OpenFin application to fail if it can't be downloaded whereas the second entry says that the OpenFin app can continue loading if it fails to fetch the asset.
+
+#### App Assets via Code
+
+**One thing to note is that in the manifest approach above the RVM will fetch these assets on first launch of your manifest (or whenever the version of your asset changes)**. This might not be the behavior you want. You might want assets to be fetched after the user logs in or you might want to have dynamic app assets based on the currently logged in user.
+
+To define and fetch an asset via code you would do the following:
+
+```js
+try {
+    // this is hardcoded here but could have come from a rest api call
+  const appAsset: OpenFin.AppAssetInfo = {
+   src: "http://filesamples.com/exe.zip",
+   alias: "myApp",
+   version: "4.12.8",
+   target: "myApp.exe",
+   args: "a b c d"
+  };
+
+  await fin.System.downloadAsset(appAsset, (progress => {
+   const downloadedPercent = Math.floor((progress.downloadedBytes / progress.totalBytes) * 100);
+   console.log(`Downloaded ${downloadedPercent}% of app asset.`);
+  }));
+  console.log("App Asset is downloaded.");
+ } catch(err) {
+  console.error("There has been an error when trying to fetch the app asset.". err);
```

Comment:
Comma not dot
`app asset.". err` => `app asset.", err`

Reply:
updated